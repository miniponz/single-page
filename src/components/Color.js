import React from 'react';
import PropTypes from 'prop-types';


// const name = 'purple';
// const hex = '#800080';
// const rgb = '128,0,128';

function compToHex(rgb) {
  let hex = rgb.toString(16);
  return hex.length == 1 ? '0' + hex : hex;
}

function rgbToHex(rgb) {
  const regex = /(\d{1,3})/g;
  const divided = rgb.match(regex);
  let r = parseInt(divided[0]);
  let g = parseInt(divided[1]);
  let b = parseInt(divided[2]);
  return '#' + compToHex(r) + compToHex(g) + compToHex(b);
}


function Color({ name, hex, rgb }) {
  return (
    <dl>
      <dt>Name</dt>
      <dd>{name}</dd>

      <dt>Hex</dt>
      <dd>{hex || rgbToHex(rgb)}</dd>

      <dt>RGB</dt>
      <dd>{rgb}</dd>
    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string.isRequired,
  hex: PropTypes.string,
  rgb: PropTypes.string.isRequired,
};


export default Color;
