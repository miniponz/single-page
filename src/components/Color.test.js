import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color Component', () => {
  it('matches the color thing', () => {
    const wrapper = shallow(<Color name="purple"  rgb="128,0,128" />);

    expect(wrapper).toMatchSnapshot();
  });
});
