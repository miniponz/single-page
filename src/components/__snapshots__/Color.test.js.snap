// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Component matches the color thing 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    purple
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    #800080
  </dd>
  <dt>
    RGB
  </dt>
  <dd>
    128,0,128
  </dd>
</dl>
`;
